src/tests/network-api-test/network-api-test.c

```diff
@@ -0,0 +1,603 @@
+
+    /* Bind peer2 socket to address */
+    actual = OS_SocketBind(p2_socket_id, &p2_addr);
+    UtAssert_True(actual == expected, "OS_SocketBind() (%ld) == OS_SUCCESS", (long)actual);
+
+    /*
+     * Test for invalid and other nominal input parameters 
+     * to the network functions being called above
+     */
+
+    /* OS_SocketOpen */
+    expected = OS_SUCCESS;
+    actual = OS_SocketOpen(&socket_id, OS_SocketDomain_INET6, OS_SocketType_DATAGRAM);
+    UtAssert_True(actual == expected, "OS_SocketOpen(NULL) (%ld) == OS_SUCCESS", (long)actual);
+
+    expected = OS_INVALID_POINTER;
+    actual = OS_SocketOpen(NULL, OS_SocketDomain_INVALID, OS_SocketType_INVALID);
+    UtAssert_True(actual == expected, "OS_SocketOpen(NULL) (%ld) == OS_INVALID_POINTER", (long)actual);
+
+    expected = OS_ERR_NOT_IMPLEMENTED;
+    actual = OS_SocketOpen(&socket_id, OS_SocketDomain_MAX, OS_SocketType_MAX);
+    UtAssert_True(actual == expected, "OS_SocketOpen() (%ld) == OS_ERR_NOT_IMPLEMENTED", (long)actual);
+
+    /* OS_SocketAddrInit */
+    expected = OS_SUCCESS;
+    actual = OS_SocketAddrInit(&addr, OS_SocketDomain_INET6);
+    UtAssert_True(actual == expected, "OS_SocketAddrInit() (%ld) == OS_SUCCESS", (long)actual);
+
+    expected = OS_INVALID_POINTER;
+    actual = OS_SocketAddrInit(NULL, OS_SocketDomain_INET6);
+    UtAssert_True(actual == expected, "OS_SocketAddrInit() (%ld) == OS_INVALID_POINTER", (long)actual);
+
+    expected = OS_ERR_NOT_IMPLEMENTED;
+    actual = OS_SocketAddrInit(&addr, OS_SocketDomain_INVALID);
+    UtAssert_True(actual == expected, "OS_SocketAddrInit() (%ld) == OS_ERR_NOT_IMPLEMENTED", (long)actual);
+
+    expected = OS_INVALID_POINTER;
+    actual = OS_SocketAddrInit(NULL, OS_SocketDomain_INVALID);
+    UtAssert_True(actual == expected, "OS_SocketAddrInit() (%ld) == OS_INVALID_POINTER", (long)actual);
+
+    /* OS_SocketAddrSetPort */
+    expected = OS_ERR_BAD_ADDRESS;
+    actual = OS_SocketAddrSetPort(&addr, 0xFFFF);
+    UtAssert_True(actual == expected, "OS_SocketAddrSetPort() (%ld) == OS_ERR_BAD_ADDRESS", (long)actual);
+
+    expected = OS_INVALID_POINTER;
+    actual = OS_SocketAddrSetPort(NULL, 1234);
+    UtAssert_True(actual == expected, "OS_SocketAddrSetPort() (%ld) == OS_INVALID_POINTER", (long)actual);
+
+    /* OS_SocketAddrFromString */
+    expected = OS_INVALID_POINTER;
+    actual = OS_SocketAddrFromString(NULL, "127.0.0.1");
+    UtAssert_True(actual == expected, "OS_SocketAddrFromString() (%ld) == OS_SUCCESS", (long)actual);
+
+    /* OS_SocketBind */
+    expected = OS_ERR_INVALID_ID;
+    actual = OS_SocketBind(1, &addr);
+    UtAssert_True(actual == expected, "OS_SocketBind() (%ld) == OS_ERR_INVALID_ID", (long)actual);
+
+    expected = OS_ERR_INCORRECT_OBJ_STATE;
+    memset(&inv_addr,0,sizeof(inv_addr));
+    actual = OS_SocketBind(p2_socket_id, &inv_addr);
+    UtAssert_True(actual == expected, "OS_SocketBind() (%ld) == OS_ERR_INCORRECT_OBJ_STATE", (long)actual);
+
+    OS_close(socket_id);
```

Comment:
This `OS_close` should be earlier - up around line 120 (i.e. right after it was opened, since it isn't used anywhere thereafter, and could be overwritten in line 126).